app/lib/user.rb

```diff
@@ -0,0 +1,27 @@
+require 'bcrypt'
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String, unique: true, message: 'This email is already taken'
+  property :name, String
+  property :user_name, String
+  property :password_digest, Text
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    user if user && BCrypt::Password.new(user.password_digest) == password
+  end
+
+end
```

Comment:
I see that there is no association between message and user... there is something but it is done manually, instead of using DataMapper resources.

The message can belong only to one user and the user can have many messages ... this will add that user, and better, the user_id to the message, which will create the association.